File: app/api/gogoreco/entities/comment.rb
module Gogoreco
  module Entities
    class Comment < Grape::Entity
      expose :pretty_id, as: :id
      expose :content, if: lambda {|s,o| o[:entity_options]["comment"][:content]}
      expose :related_evaluation, using: Gogoreco::Entities::Evaluation, if: lambda{|s,o| o[:entity_options]["comment"][:related_evaluation]}
      expose :author, using: Gogoreco::Entities::User, if: lambda{|s,o|, o[:entity_options]["comment"][:author]}
    end
  end
end

File: FrontApp/src/common/resources/comment.js
angular.module('resources.comment', [
  'restangular'
])

.factory('Comment', ['Restangular', function(Restangular){

  var Comment = {
    get: function( id ){
      var params = {
        entities: {
          comment: {
            "author": true,
            "content": true,
            "id": true,
            "related_evaluation": true
          },
          user: {
            name: true,
            image: true
          }
        }
      };
      return Restangular.one('comments', id ).customPOST( params );
    }
  };

  return Comment;

}]);
